### DISCUSSION TITLE
日本語版の記事がすべて markdown 化されます

### DISCUSSION THREAD

**Discussion by mfuji09** [mozilla-japan/translation#622] [General]
MDN では、翻訳記事の markdown 化を急ぐことになりました。日本語版の記事も近日中にすべて、スクリプトを使用して markdown 化されることになりました。
HTML のファイルが残っていることが、新機能開発の障害になっているためだそうです。

そのため、以下のことをお知らせしておきます。

- markdown 化の準備中・作業中に控えるべきこと
  - .html 形式の記事の編集・追加は控えてください。（変換後に競合となる可能性があるため）
  - .html から .md への変換も控えてください。
- markdown 化の準備中・作業中に行っても大丈夫なこと
  - .md 形式の記事を修正しても大丈夫です。
  - 未翻訳記事を新規に（.md で）翻訳しても大丈夫です。

なお、現在レビュー待ちになっているプルリクエストについては、この間にできるだけ早くレビュー（およびマージ）を行いますが、レビューの修正依頼に対して反応がない場合は、レビュアーが修正するかクローズする可能性があります。

何かご意見などがあればコメントしてください。

**Comment by debiru** [comment]
一括 markdown 化にあたって気になる点を示しておきます。

- index.html と index.md が同時に存在するディレクトリがある（現時点で3件）
    - https://mdn.lavoscore.org/?file=dup
    - 直近のコミットで一部対応されている
        - https://github.com/mdn/translated-content/commit/2d44ac7d576feaab62deaf62bd32b05b1b54d211
    - `[translated-content] $ find files/ja -name 'index.html' -or -name 'index.md' | sed -E 's/\.(html|md)$//' | sort | uniq -d`
        - /web/api/window/blur_event
        - /web/api/window/focus_event
        - /web/javascript/reference/global_objects/symbol
- 何らかの理由で、日本語版にのみ記事が存在するページがある（現時点で281件、今後も増減する可能性がある）
    - https://mdn.lavoscore.org/?filter=not-en&file=html

上記に関して、不要と思われるファイルを事前に削除する必要があるかもしれません。

**Reply by mfuji09** [reply]
ありがとうございます。

.html と .md のファイルが同時に存在する件は、既にPRが出ているのですが、複数言語にまたがっているため承認が難しいようです。新しく PR を作成して、変換前に3つのファイルを削除しておいた方がよさそうですね。

日本語版のみに存在するページのうち、索引ページについては削除してもよさそうですね。

両方とも、PRを作成して削除しておきます。

**Comment by debiru** [comment]
作業状況はこのあたりでしょうか。

- トピック [/mdn/translated-content/discussions/1236](https://github.com/mdn/translated-content/discussions/1236)
- ja-PR https://github.com/mdn/translated-content/pull/7628

**Reply by mfuji09** [reply]
前回、Yariプラットフォームへの移行の際には詳細な説明をいただくことができたのですが、今回のmdへの移行については事前の情報が何もないので、作業状況がどうなるのかがよく見えません。
今回作業している人が新人のようで、オープンソースコミュニティに不慣れだと言っていました。

**Reply by debiru** [reply]
翻訳作業（更新）をしようと思ったのですが、対象ページが html だったので手が止まってしまっています。

Markdown化プロジェクトの進捗の見通しがちょっと悪そうですね。（いつ完了するか分からない）

**Comment by mfuji09** [comment]
ko や es などのロケールでは、ロケールのメンテナーが独自に変換作業を始めているようですので、 ja ロケールでも始めてしまいましょうか。（変換スクリプトの利用方法は把握しました。）

私が変換作業を行えば、コミットのタイミングをコントロールできるので、いつ変換作業が行われるかを待たなくてもよくなります。

ただ、変換時の不具合もあると思いますので、すべてを一括して行うのは怖いと思います。
もし、先に変換してほしいページがありましたら、ディレクトリ単位くらいで指定していただければ、変換時の問題の把握も兼ねて、試しに変換してみようと思います。

（ちなみに、私の手元で、試しに mdn ディレクトリ以下の記事を変換してみています）。

**Reply by potappo** [reply]
今日は夕方までずっと移動なので、ディレクトリのリクエストなど細かいコメントは出来ませんが、ja コミュニティの方で変換できるならその方が良いと思いますので、ぜひお願いします。

**Comment by mfuji09** [comment]
変換作業を行います。変換の進捗状況については、下記の issue をご覧ください。
https://github.com/mdn/translated-content/issues/7782

**Comment by potappo** [comment]
まともにレビューするのでは追いつきませんので、問題が判明次第、その解決のための issue を立てようかと思います。
現在判明している問題は以下です。
 - Note ブロック
 以下は日本語版が英語版の更新に追いついていないことが原因ですが、別途 issue を立てて対策した方が良いと思います。
 - BCD クエリ―のエラー
 - 画像が表示されなくなっている

**Comment by potappo** [comment]
ひとまず立てました。
- https://github.com/mdn/translated-content/issues/7848

- https://github.com/mdn/translated-content/issues/7850

- https://github.com/mdn/translated-content/issues/7849
